components/MapComponent.tsx

```diff
@@ -23,83 +32,142 @@ interface Props<T> {
     longitudeDelta: number
   }
 
-  markers: mapMarker[]
+  initialRadius: {
+    radius: number
+  }
+
+  markers: MapMarker[]
+  customizers: MarkerCustomize[]
+
+  currentSelectedMarker: String | undefined
 
   movementSettings: {
     canScroll: boolean
     canZoom: boolean
     canRotate: boolean
   }
+}
 
-  extractKey: (event: any) => string
+// When clicking on the marker, zoom towards where it is.
+// Using stackOverflow formula
+// stackoverflow.com/questions/5206018/fit-mapview-to-circle-bounds
+function circleCoordinateGeneration (lat: number, long: number, radius: number) {
+  const earthRadius = 6378.1 // km
```

Comment:
Btw we actually also have a `EARTH_RADIUS_METERS` constant in the same file as `degreesToRadians`, so you could just use that and divide by 1000. It could also remove the small `// km` comment that you have.